### PR TITLE
server/core: fix DiffRegionKeyInfo no use escaped key

### PR DESCRIPTION
**Pull request by nolouch**
## What have you changed? (required)

No use of escape characters will print strange logs. this PR fix it.
## What are the type of the changes (required)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (required)?

unit test.

### CODE REVIEW

File: server/core/region.go [hunk 1]
```diff
@@ -816,14 +816,14 @@ func DiffRegionKeyInfo(origin *RegionInfo, other *RegionInfo) string {
 		otherKey := &metapb.Region{StartKey: other.Region.StartKey}
 		ret = append(ret, fmt.Sprintf("StartKey Changed:{%s} -> {%s}", originKey, otherKey))
 	} else {
-		ret = append(ret, fmt.Sprintf("StartKey:{%s}", origin.Region.StartKey))
+		ret = append(ret, fmt.Sprintf("StartKey:{%s}", &metapb.Region{EndKey: origin.Region.StartKey}))
```
Comment (rleungx) [inline server/core/region.go]: EndKey?

File: server/core/region_test.go [hunk 2]
```diff
@@ -97,3 +99,31 @@ func (s *testRegionMapSuite) check(c *C, rm *regionMap, ids ...uint64) {
...
+type testRegionKey struct{}
+
+func (*testRegionKey) TestRegionKey(c *C) {
+	testCase := []struct {
+		key    string
+		expect string
+	}{
+		{`"t\x80\x00\x00\x00\x00\x00\x00\xff!_r\x80\x00\x00\x00\x00\xff\x02\u007fY\x00\x00\x00\x00\x00\xfa"`,
+			`"t\200\000\000\000\000\000\000\377!_r\200\000\000\000\000\377\002\177Y\000\000\000\000\000\372"`},
+		{"\"\\x80\\x00\\x00\\x00\\x00\\x00\\x00\\xff\\x05\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\xf8\"",
```
Comment (disksing) [inline server/core/region_test.go]: Why not use ` this line? So many "\\"s.

File: server/core/region_test.go [hunk 3]
```diff
@@ -97,3 +99,40 @@ func (s *testRegionMapSuite) check(c *C, rm *regionMap, ids ...uint64) {
...
+			`"\200\000\000\000\000\000\000\377\005\000\000\000\000\000\000\000\370"`},
+	}
+	got := make([]byte, 30)
+	for _, t := range testCase {
+		_, err := fmt.Sscanf(t.key, "%q", &got)
```
Comment (disksing) [inline server/core/region_test.go]: Can we use `strings.Unquote` instead?

**Comment by nolouch** [comment]
PTAL @disksing @rleungx. This change should be merged before we release rc1.

**Comment by disksing** [comment]
PTAL @rleungx

**Comment by liukun4515** [comment]
LGTM

**Pull request merged**